ci-operator/step-registry/workers-rhel/ibmcloud-provision/workers-rhel-ibmcloud-provision-commands.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -xeuo pipefail
+
+# IBM Cloud CLI login
+function ibmcloud_login {
+  echo "Try to login..."
+  "${IBMCLOUD_CLI}" login -r ${LEASED_RESOURCE} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+}
+
+function checkCli() {
+  export IBMCLOUD_CLI=ibmcloud
+  export IBMCLOUD_HOME=/output
+  echo "check IBMCLOUD_CLI: ${IBMCLOUD_CLI}..."
+  echo "Pause for debug...."
+  command -v ${IBMCLOUD_CLI}
+  ${IBMCLOUD_CLI} --version
+  ${IBMCLOUD_CLI} plugin list
+}
+
+checkCli
+region="${LEASED_RESOURCE}"
+export region
+
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+
+export SSH_PRIV_KEY_PATH=${CLUSTER_PROFILE_DIR}/ssh-privatekey
+export PULL_SECRET_PATH=${CLUSTER_PROFILE_DIR}/pull-secret
+export SSH_PUB_KEY_PATH=${CLUSTER_PROFILE_DIR}/ssh-publickey
+
+# Get existing vnet info
+VPC_CONFIG="${SHARED_DIR}/customer_vpc_subnets.yaml"
+if [[ ! -f "${VPC_CONFIG}" ]]; then
+  echo "Fail to find of VPC info file ${VPC_CONFIG}, abort." && exit 1
+fi
+echo "Reading variables from ${VPC_CONFIG}..."
+vpcName=$(yq-go r "${VPC_CONFIG}" 'platform.ibmcloud.vpcName')
+resource_group=$(yq-go r "${VPC_CONFIG}" 'platform.ibmcloud.resourceGroupName')
+infra_id=$(jq -r .infraID ${SHARED_DIR}/metadata.json)
+echo "Using region: ${region}  resource_group: ${resource_group} vpc: ${vpcName}"
+
+ibmcloud_login
+
+${IBMCLOUD_CLI} target -g ${resource_group} -r ${region}
+
+workdir=`mktemp -d`
+
+# Get bastion RG info
+vnet_RG=$(yq-go r "${VPC_CONFIG}" 'platform.ibmcloud.networkResourceGroupName')
+echo "vpc RG: ${vnet_RG}"
+
+#when run more jobs at the same time, fail to create key with the same fingerprint, so create the key (ci-qe-key) in Default group for all test. 
+${IBMCLOUD_CLI} is keys --all-resource-groups
+
+preDefaultKey="ci-qe-key"
```

Comment:
Now I add the keyname into vault, you can get it by `rhel-ssh-keys=${CLUSTER_PROFILE_DIR}/ibmcloud-sshkey-name`

Reply:
thanks a lot, I will put my changes when [4.13-amd64-nightly-ibmcloud-ipi-workers-rhel8](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/39088/rehearse-39088-periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-ibmcloud-ipi-workers-rhel8-p2-f14/1656473915871989760) test done